### PR TITLE
(DOCSP-30804): Flutter: Break out CRUD section, List and Set asResults()

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-30804

- [CRUD/Read](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-30804/sdk/flutter/crud/read/#convert-lists-or-sets-to-results): New "Convert Lists or Sets to Results" section with tested, Bluehawked code example demonstrating the functionality.

As part of this PR, I also broke out the Read & Write Data page into a new CRUD section. I put it at the root of the SDK because a future iteration will remove the "Realm Database" docs from the "Realm Database" section of the SDK. Putting it at the root of the SDK means we won't have to make redirects for this section when that time comes - we'll only have to update the ToC.

I copied language that has already been approved in the C++ and Kotlin SDKs to flesh out CRUD details. The only changes I made to that language was to update the SDK docs links referenced, and to omit Realm naming when possible in favor of the new naming/positioning.

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [x] Did you tag pages appropriately?
  - genre
  - meta.keywords
  - meta.description
- [x] Describe your PR's changes in the Release Notes section
- [x] Create a Jira ticket for related docs-app-services work, if any

### Release Notes

- **Flutter SDK**
  - Realm/Realm Database/Read & Write Data: Break out the single page to a new CRUD section.
  - Realm/Realm Database/CRUD/Read: New "Convert Lists or Sets to Results" section with tested, Bluehawked code example demonstrating the functionality.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/flutter/crud.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+.. toctree::
+   :titlesonly:
+
+   Create </sdk/flutter/crud/create>
+   Read </sdk/flutter/crud/read>
+   Update </sdk/flutter/crud/update>
+   Delete </sdk/flutter/crud/delete>
+
+- :doc:`Create </sdk/flutter/crud/create>`
+- :doc:`Create </sdk/flutter/crud/read>`
```
Comment (cbullinger) [inline source/sdk/flutter/crud.txt:14]:
```suggestion
- :doc:`Read </sdk/flutter/crud/read>`
```

File: config/redirects [hunk 2]
```diff
@@ -1277,3 +1277,7 @@ raw: ${prefix}/sdk/kotlin/realm-database/database-internals -> ${base}/introduct
 raw: ${prefix}/sdk/node/realm-database -> ${base}/introduction/
 raw: ${prefix}/sdk/react-native/realm-database -> ${base}/introduction/
 raw: ${prefix}/sdk/swift/realm-database -> ${base}/introduction/
+
+# Flutter swarm/parity
+
+raw: ${prefix}/sdk/flutter/realm-database/read-and-write-data -> ${base}/sdk/flutter/realm-database/crud/create
```
Comment (cbullinger) [inline config/redirects:1283]:
```suggestion
raw: ${prefix}/sdk/flutter/realm-database/read-and-write-data -> ${base}/sdk/flutter/realm-database/crud/create/
```

File: examples/dart/test/read_write_data_test.dart [hunk 3]
```diff
@@ -78,6 +79,43 @@ void main() {
...
+        Person(ObjectId(), 'Luke', hobbies: [
+          'Going to Tashi Station',
+          'Fixing the Moisture Vaporators'
+        ]),
+        Person(ObjectId(), 'Leia', hobbies: [
+          'Going on diplomatic missions',
+          'Rescuing short stormtroopers'
+        ]),
+        Person(ObjectId(), 'Han',
+            hobbies: ['Shooting first', 'Making fast Kessel Runs']),
```
Comment (cbullinger) [inline examples/dart/test/read_write_data_test.dart:96]: reformat for consistency with the others?
```
Person(ObjectId(), 'Han', hobbies: [
   'Shooting first', 
   'Making fast Kessel Runs'
]),
```

File: examples/dart/test/read_write_data_test.dart [hunk 4]
```diff
@@ -78,6 +79,43 @@ void main() {
...
+        Person(ObjectId(), 'Chewbacca', hobbies: [
+          'Fixing the Millenium Falcon',
+          'Tearing the arms off of droids'
+        ])
+      ]);
+      realm.write(() => realm.add(heroes));
+
+      // This turns the Team object's 'crew' List into a RealmResults<Person>.
```
Comment (cbullinger) [inline examples/dart/test/read_write_data_test.dart:104]:
```suggestion
      // Converts the Team object's 'crew' List into a RealmResults<Person>.
```

File: examples/dart/test/read_write_data_test.dart [hunk 5]
```diff
@@ -78,6 +79,43 @@ void main() {
...
+      final heroesCrewAsResults = heroes.crew.asResults();
+
+      final luke = heroesCrewAsResults.query("name == 'Luke'").first;
+      // This turns Luke's 'hobbies' list into a RealmResults<String>
```
Comment (cbullinger) [inline examples/dart/test/read_write_data_test.dart:108]:
```suggestion
      // Converts Luke's 'hobbies' list into a RealmResults<String>
```

File: examples/dart/test/read_write_data_test.dart [hunk 6]
```diff
@@ -78,6 +79,43 @@ void main() {
...
+            hobbies: ['Shooting first', 'Making fast Kessel Runs']),
+        Person(ObjectId(), 'Chewbacca', hobbies: [
+          'Fixing the Millenium Falcon',
+          'Tearing the arms off of droids'
+        ])
+      ]);
+      realm.write(() => realm.add(heroes));
+
+      // This turns the Team object's 'crew' List into a RealmResults<Person>.
+      final heroesCrewAsResults = heroes.crew.asResults();
```
Comment (cbullinger) [inline examples/dart/test/read_write_data_test.dart:105]: would it be helpful to emphasize this and line 109?

**Review by cbullinger — APPROVED** [review]
a couple suggestions, but overall LGTM once the staging build confirms the code example is displaying correctly

File: config/redirects (hunk 2 above)
Comment (dacharyc) [reply config/redirects]: Blah, good catch, thank you!

File: examples/dart/test/read_write_data_test.dart (hunk 3 above)
Comment (dacharyc) [reply examples/dart/test/read_write_data_test.dart:96]: I couldn't do that even if I tried - this is all the linter, baybee. Upside/downside of using a linter.

File: source/sdk/flutter/crud.txt (hunk 1 above)
Comment (dacharyc) [reply source/sdk/flutter/crud.txt]: Bah, where was my attention to detail? Thank you for catching this!

**Pull request merged**